src/java/org/httpkit/HttpUtils.java

```diff
@@ -455,7 +455,7 @@ public static ByteBuffer[] HttpEncode(int status, HeaderMap headers, Object body
                 if (bodyBuffer != null) {
                     // trust the computed length
                     headers.putOrReplace(CONTENT_LENGTH, Integer.toString(bodyBuffer.remaining()));
-                } else {
+                } else if ((status / 100) != 1 && status != 204) {
```

Comment:
Disclaimer: I am no Java expert nor very proficient in the HTTP protocols and servers or the http-kit code in general

From the RFC:
> A server MUST NOT send a Content-Length header field in any response 
>  with a status code of 1xx (Informational) or 204 (No Content).

If for some reason `bodyBuffer` would be non-`null` the code would still send content length for status codes 1xx and 204. :shrug: 

Consider the following to make the RFC implementation more explicit:
```java
if ((status / 100) != 1 && status != 204) {
    // trust the computed length
    headers.putOrReplace(
        CONTENT_LENGTH,
        (bodyBuffer != null)) Integer.toString(bodyBuffer.remaining()) : "0"
    );
}
```

really brilliant with the `(status / 100) != 1` btw.

Reply:
Thanks, comitted a new change based on your suggestion. 

Unfortunately, I cannot test the new commits on my end because I am currently away from my workstation (although I don't see any problem with this change). If anyone can test it would be great (@restenb?).

Reply:
I don't think the RFC mentions the case of a 101 response having a body, but it would be unnecessary and perhaps unexpected to receive a body in a 101 response, no? If the request has a body, that introduces the need to think about setting `CONTENT_LENGTH` again.

I think the original implementation was probably fine, and servers would be incorrect in sending a 101 response with a body.

Reply:
@davidhjp01 Hi David, thanks for your PR! I'm prepping for the upcoming http-kit v2.7 release and would love to get this included.

What's the current status of the PR, after @restenb's [last comment](https://github.com/http-kit/http-kit/pull/499#discussion_r1045516042)?

Thanks!

Reply:
I changed it back to the original version. We can also omit `bodyBuffer` if the passed status is 1xx or 204 but since this is an internal API I guess this is not much an issue (i.e. knowing that we should not pass non-null `body` with 1xx or 204).